Content.Server/_NF/GameRule/NfAdventureRuleSystem.cs

```diff
@@ -108,240 +118,275 @@ private void OnPlayerSpawningEvent(PlayerSpawnCompleteEvent ev)
-            {
-                Offset = -tinniaOffset
-            }))
-        {
-            var meta = EnsureComp<MetaDataComponent>(churchUids[0]);
-            _meta.SetEntityName(churchUids[0], "Omnichurch Beacon", meta);
-            _shuttle.SetIFFColor(churchUids[0], factionColor);
-        }
+        var dungenTypes = _prototypeManager.EnumeratePrototypes<DungeonConfigPrototype>();
 
-        if (_map.TryLoad(mapId, lpbravo, out var lpbravoUids, new MapLoadOptions
-            {
-                Offset = _random.NextVector2(2150f, 4850f)
-            }))
+        foreach (var dunGen in dungenTypes)
         {
-            if (_prototypeManager.TryIndex<GameMapPrototype>("LPBravo", out var stationProto))
+            var seed = _random.Next();
+            var offset = GetRandomPOICoord(3000f, 8500f, true);
+            if (!_map.TryLoad(_mapId, "/Maps/_NF/Dungeon/spaceplatform.yml", out var grids,
+                    new MapLoadOptions
+                    {
+                        Offset = offset
+                    }))
             {
-                _station.InitializeNewStation(stationProto.Stations["LPBravo"], lpbravoUids);
+                continue;
             }
 
-            var meta = EnsureComp<MetaDataComponent>(lpbravoUids[0]);
-            _meta.SetEntityName(lpbravoUids[0], "Listening Point Bravo", meta);
-            _shuttle.SetIFFColor(lpbravoUids[0], lpbravoColor);
-            _shuttle.AddIFFFlag(lpbravoUids[0], IFFFlags.HideLabel);
-        }
+            var mapGrid = EnsureComp<MapGridComponent>(grids[0]);
+            _shuttle.AddIFFFlag(grids[0], IFFFlags.HideLabel);
+            _console.WriteLine(null, $"dungeon spawned at {offset}");
+            offset = new Vector2i(0, 0);
 
-        // if (_map.TryLoad(mapId, northpole, out var northpoleUids, new MapLoadOptions
-        //     {
-        //         Offset = _random.NextVector2(2150f, 3900f)
-        //     }))
-        // {
-        //     var meta = EnsureComp<MetaDataComponent>(northpoleUids[0]);
-        //     _shuttle.SetIFFColor(northpoleUids[0], lpbravoColor);
-        //     _shuttle.AddIFFFlag(northpoleUids[0], IFFFlags.HideLabel);
-        // }
-
-        if (_map.TryLoad(mapId, arena, out var depotUid5s, new MapLoadOptions
-            {
-                Offset = _random.NextVector2(2200f, 4200f)
-            }))
-        {
-            var meta = EnsureComp<MetaDataComponent>(depotUid5s[0]);
-            _meta.SetEntityName(depotUid5s[0], "The Pit", meta);
-            _shuttle.SetIFFColor(depotUid5s[0], civilianColor);
+            //pls fit the grid I beg, this is so hacky
+            //its better now but i think i need to do a normalization pass on the dungeon configs
+            //because they are all offset. confirmed good size grid, just need to fix all the offsets.
+            _dunGen.GenerateDungeon(dunGen, grids[0], mapGrid, (Vector2i) offset, seed);
```

Comment:
Remove offset assignment (2/2)
```suggestion
            _dunGen.GenerateDungeon(dunGen, grids[0], mapGrid, new Vector2i(0, 0), seed);
```